bin/stacks/rpc-gateway-fallback-stack.ts

```diff
@@ -0,0 +1,147 @@
+import * as cdk from 'aws-cdk-lib'
+import { aws_cloudwatch, aws_cloudwatch_actions, aws_iam, aws_lambda, aws_lambda_nodejs } from 'aws-cdk-lib'
+import { Construct } from 'constructs'
+import path from 'path'
+import { getRpcGatewayEnabledChains } from '../../lib/rpc/ProdConfig'
+import { ComparisonOperator, MathExpression } from 'aws-cdk-lib/aws-cloudwatch'
+import { RetentionDays } from 'aws-cdk-lib/aws-logs'
+import { DynamoDBTableProps } from './routing-database-stack'
+import * as aws_dynamodb from 'aws-cdk-lib/aws-dynamodb'
+
+export interface RpcGatewayFallbackStackPros extends cdk.NestedStackProps {
+  rpcProviderHealthStateDynamoDb: aws_dynamodb.Table
+}
+
+export class RpcGatewayFallbackStack extends cdk.NestedStack {
+  constructor(scope: Construct, name: string, props: RpcGatewayFallbackStackPros) {
+    super(scope, name, props)
+
+    const lambdaRole = new aws_iam.Role(this, 'ProviderFallbackLambdaRole', {
+      assumedBy: new aws_iam.ServicePrincipal('lambda.amazonaws.com'),
+      managedPolicies: [
+        aws_iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaBasicExecutionRole'),
+        aws_iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaRole'),
+        aws_iam.ManagedPolicy.fromAwsManagedPolicyName('CloudWatchLambdaInsightsExecutionRolePolicy'),
+        aws_iam.ManagedPolicy.fromAwsManagedPolicyName('AWSXRayDaemonWriteAccess'),
+      ],
+    })
+
+    const rpcHealthProviderStateDynamoDB = props.rpcProviderHealthStateDynamoDb
+    rpcHealthProviderStateDynamoDB.grantReadWriteData(lambdaRole)
+
+    const region = cdk.Stack.of(this).region
+
+    const providerFallbackLambda = new aws_lambda_nodejs.NodejsFunction(this, 'ProviderFallbackLambda', {
+      role: lambdaRole,
+      runtime: aws_lambda.Runtime.NODEJS_18_X,
+      entry: path.join(__dirname, '../../lib/rpc/handler/index.ts'),
+      handler: 'fallbackHandler',
+      timeout: cdk.Duration.seconds(5),
+      memorySize: 1024,
+      description: 'Provider Fallback Lambda',
+      bundling: {
+        minify: true,
+        sourceMap: true,
+      },
+      layers: [
+        aws_lambda.LayerVersion.fromLayerVersionArn(
+          this,
+          'InsightsLayer',
+          `arn:aws:lambda:${region}:580247275435:layer:LambdaInsightsExtension:14`
+        ),
+      ],
+      tracing: aws_lambda.Tracing.ACTIVE,
+      logRetention: RetentionDays.ONE_MONTH,
```

Comment:
Do you actually need one month of log retention for the fallback lambda? It's a stateless lambda, in a sense that it persists the provider health into DDB, so in case something goes wrong with this lambda, it feels like we should be able to debug within a month.